p2p/src/net/mock/backend.rs

```diff
@@ -208,21 +238,30 @@ where
         }
     }
 
-    /// Send request to remote peer
+    /// Sends a request to the remote peer.
     async fn send_request(
         &mut self,
-        peer_id: &MockPeerId,
+        peer_id: MockPeerId,
         request: message::Request,
     ) -> crate::Result<MockRequestId> {
         let peer = self
             .peers
-            .get_mut(peer_id)
+            .get_mut(&peer_id)
             .ok_or(P2pError::PeerError(PeerError::PeerDoesntExist))?;
 
-        let (request_id, request) = self.request_mgr.make_request(peer_id, request)?;
-
+        let (request_id, request) = self.request_mgr.make_request(request)?;
         peer.tx.send(MockEvent::SendMessage(request)).await.map_err(P2pError::from)?;
 
+        let is_inserted = self.pending_responses.insert(request_id);
+        debug_assert!(is_inserted);
+
+        let timeout = self.p2p_config.request_timeout.clone().into();
+        let sender = self.timeouts_sender.clone();
+        tokio::spawn(async move {
+            time::sleep(Duration::from_secs(timeout)).await;
+            let _ = sender.send((request_id, peer_id));
+        });
```

Comment:
Would it be worth it to somehow kill the timeout task once the response is received so it does not hang around for the full timeout period?

Reply:
I haven’t been able to come up with a solution how to do that properly. I'm open to suggestions. 🙃

Reply:
Hmm not sure. Maybe add the task handle to `pending_responses` and call `Task::abort` when the request is resolved. Would that work?

Reply:
Good idea, I will try that!